modules/carto/src/layers/carto-layer.js

```diff
@@ -20,7 +20,13 @@ const defaultProps = {
   /* API v3 PARAMETERS */
   /**********************/
   // (String, required): connection name at CARTO platform
-  connection: null
+  connection: null,
+
+  // (String, optional): name of geo_column in CARTO platform
+  geoColumn: null,
+
+  // (Array<String>, optional): names of columns to fetch
+  columns: null
```

Comment:
If you declare `type: 'array'` here then the layer diff will block unnecessary updates unless the prop changes deeply